smoketest/scripts/cli/test_vpn_ipsec.py

```diff
@@ -0,0 +1,123 @@
+#
+# Copyright (C) 2020 VyOS maintainers and contributors
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 or later as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import unittest
+
+from base_vyostest_shim import VyOSUnitTestSHIM
+
+from vyos.util import process_named_running
+
+tunnel_path = ['interfaces', 'tunnel']
+nhrp_path = ['protocols', 'nhrp']
+base_path = ['vpn', 'ipsec']
+
+ipsec_conf = "/etc/ipsec.conf"
+ipsec_secrets = "/etc/ipsec.secrets"
+swanctl_conf = "/etc/swanctl/swanctl.conf"
+
+class TestVPNIPsec(VyOSUnitTestSHIM.TestCase):
+    def tearDown(self):
+        # Delete vpn openconnect configuration
+        self.cli_delete(base_path)
+        self.cli_commit()
+
+    def test_site_to_site(self):
+        self.cli_delete(base_path)
+        self.cli_delete(tunnel_path)
+
+        # Tunnel
+        self.cli_set(tunnel_path + ["tun0", "encapsulation", "gre"])
+        self.cli_set(tunnel_path + ["tun0", "local-ip", "192.0.2.10"])
+        self.cli_set(tunnel_path + ["tun0", "remote-ip", "203.0.113.45"])
+        self.cli_set(tunnel_path + ["tun0", "address", "10.10.10.1/30"])
+
+        # IKE/ESP Groups
+        self.cli_set(base_path + ["esp-group", "MyESPGroup", "proposal", "1", "encryption", "aes128"])
+        self.cli_set(base_path + ["esp-group", "MyESPGroup", "proposal", "1", "hash", "sha1"])
+        self.cli_set(base_path + ["ike-group", "MyIKEGroup", "proposal", "1", "dh-group", "2"])
+        self.cli_set(base_path + ["ike-group", "MyIKEGroup", "proposal", "1", "encryption", "aes128"])
+        self.cli_set(base_path + ["ike-group", "MyIKEGroup", "proposal", "1", "hash", "sha1"])
+
+        # Site to site
+        self.cli_set(base_path + ["ipsec-interfaces", "interface", "eth0"])
+        self.cli_set(base_path + ["site-to-site", "peer", "203.0.113.45", "authentication", "mode", "pre-shared-secret"])
+        self.cli_set(base_path + ["site-to-site", "peer", "203.0.113.45", "authentication", "pre-shared-secret", "MYSECRETKEY"])
+        self.cli_set(base_path + ["site-to-site", "peer", "203.0.113.45", "ike-group", "MyIKEGroup"])
+        self.cli_set(base_path + ["site-to-site", "peer", "203.0.113.45", "default-esp-group", "MyESPGroup"])
+        self.cli_set(base_path + ["site-to-site", "peer", "203.0.113.45", "local-address", "192.0.2.10"])
+        self.cli_set(base_path + ["site-to-site", "peer", "203.0.113.45", "tunnel", "1", "protocol", "gre"])
+
+        self.cli_commit()
+
+        # TODO: verify output to config files
```

Comment:
Can you please add some output verification steps which checks that the intended config line is actually rendered?Something like: https://github.com/vyos/vyos-1x/blob/current/smoketest/scripts/cli/test_system_conntrack.py#L232-L235